### PR TITLE
feat: OBS-489 - allow change tls verification by setting env variable

### PR DESCRIPTION
**Pull request by leoparente**

### CODE REVIEW

File: diode-server/netboxdiodeplugin/client.go [hunk 1]
```diff
@@ -96,7 +101,24 @@ type Client struct {
 
 // NewClient creates a new NetBox Diode plugin client
 func NewClient(logger *slog.Logger, apiKey string) (*Client, error) {
-	rt, err := newAPIRoundTripper(apiKey, http.DefaultTransport)
+
```
Comment (mfiedorowicz) [inline diode-server/netboxdiodeplugin/client.go]: nit: redundant empty line

File: diode-server/netboxdiodeplugin/client_test.go [hunk 2]
```diff
@@ -187,13 +210,23 @@ func TestRetrieveObjectState(t *testing.T) {
...
 		{
 			name:               "invalid server response",
 			params:             netboxdiodeplugin.RetrieveObjectStateQueryParams{ObjectType: netbox.DcimDeviceObjectType, ObjectID: 1},
 			apiKey:             "barfoo",
 			mockServerResponse: ``,
+			tlsSkipVerify:      true,
+			shouldError:        true,
+		},
+		{
+			name:               "tls bad ceritificate",
```
Comment (mfiedorowicz) [inline diode-server/netboxdiodeplugin/client_test.go]: typo: s/ceritificate/certificate

File: diode-server/netboxdiodeplugin/client.go [hunk 3]
```diff
@@ -96,7 +101,24 @@ type Client struct {
...
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: skipTLS(),
+		},
+	}
```
Comment (mfiedorowicz) [inline diode-server/netboxdiodeplugin/client.go]: To have it tested in more clear fashion, could we extract it into a function, i.e. `NewHTTPTransport()` and have a dedicated test asserting something like this?

```go
transport := netboxdiodeplugin.NewHTTPTransport()
assert.Equal(t, tt.expectedInsecure, transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify)
```

**Review by mfiedorowicz — COMMENTED** [review]
Thanks @leoparente, I've left a few comments to consider.

File: diode-server/netboxdiodeplugin/client.go (hunk 3 above)
Comment (leoparente) [reply diode-server/netboxdiodeplugin/client.go]: done!

File: diode-server/netboxdiodeplugin/client_test.go [hunk 4]
```diff
@@ -17,6 +17,34 @@ import (
 	"github.com/netboxlabs/diode/diode-server/netboxdiodeplugin"
 )
 
+func TestTransfortSecurity(t *testing.T) {
```
Comment (mfiedorowicz) [inline diode-server/netboxdiodeplugin/client_test.go]: nit: s/TestTransfortSecurity/TestTransportSecurity

**Review by mfiedorowicz — APPROVED** [review]

**Pull request merged**